This patch duplicates upstream commit 4144d1e, which helps ensure the
"test_net_shortwrite" test suite can succeed across all platforms by
deepening the pending-connection queue of the server it creates from 0
to 1.

See the original report at
https://gitlab.nic.cz/knot/knot-dns/-/issues/693

--- a/tests/contrib/test_net_shortwrite.c
+++ b/tests/contrib/test_net_shortwrite.c
@@ -88,7 +88,7 @@ int main(int argc, char *argv[])
	int server = net_bound_socket(SOCK_STREAM, &addr, 0);
	ok(server >= 0, "server: bind socket");

-	r = listen(server, 0);
+	r = listen(server, 1);
	ok(r == 0, "server: start listening");

	struct sockaddr *sa = (struct sockaddr *)&addr;
